### PR TITLE
Disapprove leap solutions when DateTime.IsLeapYear is used

### PR DESCRIPTION
**Pull request by EarthlingRich**
For solutions to the leap exercise usage of the DateTime.IsLeapYear method is not allowed. This PR implements functionality for the leap analyzer that disapprove solution that use this method.

### CODE REVIEW

File: src/Exercism.Analyzers.CSharp/Analyzers/Leap/LeapSolution.cs [hunk 1]
```diff
@@ -7,17 +6,20 @@ namespace Exercism.Analyzers.CSharp.Analyzers.Leap
 {
     internal class LeapSolution : ParsedSolution
     {
-        public MethodDeclarationSyntax IsLeapYearMethod { get; }
-        public ParameterSyntax YearParameter { get; }
-        public ExpressionSyntax ReturnedExpression { get; }
+        public readonly ParameterSyntax YearParameter;
```
Comment (ErikSchierboom) [inline src/Exercism.Analyzers.CSharp/Analyzers/Leap/LeapSolution.cs]: Might I ask your reason for this change? I'm not opposed to it, but I'd like to hear your thoughts on it.

**Comment by ErikSchierboom** [comment]
Another great addition. I've left a small comment.

File: src/Exercism.Analyzers.CSharp/Analyzers/Leap/LeapSolution.cs (hunk 1 above)
Comment (EarthlingRich) [reply src/Exercism.Analyzers.CSharp/Analyzers/Leap/LeapSolution.cs]: I tried to make it the same as the gigasecond solution. But now I see two-fer uses the getters. So no real reason from my side.

Comment (ErikSchierboom) [reply src/Exercism.Analyzers.CSharp/Analyzers/Leap/LeapSolution.cs]: Ah. An inconsistency on my part then.

**Comment by ErikSchierboom** [comment]
Merged. Thanks!

**Pull request merged**